presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -31,58 +30,48 @@
-                Optional.ofNullable(config.getPasswordCredentialName()),
-                basicConnectionProperties(credentialProvider));
+                new Properties(),
+                credentialProvider);
     }
 
-    public static Properties basicConnectionProperties(CredentialProvider credentialProvider)
-    {
-        Properties connectionProperties = new Properties();
-        if (credentialProvider.getConnectionUser(Optional.empty()) != null) {
-            connectionProperties.setProperty("user", credentialProvider.getConnectionUser(Optional.empty()));
-        }
-        if (credentialProvider.getConnectionPassword(Optional.empty()) != null) {
-            connectionProperties.setProperty("password", credentialProvider.getConnectionPassword(Optional.empty()));
-        }
-        return connectionProperties;
-    }
-
-    public DriverConnectionFactory(Driver driver, String connectionUrl, Optional<String> userCredentialName, Optional<String> passwordCredentialName, Properties connectionProperties)
+    public DriverConnectionFactory(Driver driver, String connectionUrl, Properties connectionProperties, CredentialProvider credentialProvider)
     {
         this.driver = requireNonNull(driver, "driver is null");
         this.connectionUrl = requireNonNull(connectionUrl, "connectionUrl is null");
         this.connectionProperties = new Properties();
         this.connectionProperties.putAll(requireNonNull(connectionProperties, "basicConnectionProperties is null"));
-        this.userCredentialName = requireNonNull(userCredentialName, "userCredentialName is null");
-        this.passwordCredentialName = requireNonNull(passwordCredentialName, "passwordCredentialName is null");
+        this.credentialProvider = requireNonNull(credentialProvider, "credentialProvider is null");
     }
 
     @Override
     public Connection openConnection(JdbcIdentity identity)
             throws SQLException
     {
-        userCredentialName.ifPresent(credentialName -> setConnectionProperty(connectionProperties, identity.getExtraCredentials(), credentialName, "user"));
-        passwordCredentialName.ifPresent(credentialName -> setConnectionProperty(connectionProperties, identity.getExtraCredentials(), credentialName, "password"));
-
-        Connection connection = driver.connect(connectionUrl, connectionProperties);
+        Properties properties = updateConnectionProperty(connectionProperties, credentialProvider, identity);
+        Connection connection = driver.connect(connectionUrl, properties);
         checkState(connection != null, "Driver returned null connection");
         return connection;
     }
 
-    private static void setConnectionProperty(Properties connectionProperties, Map<String, String> extraCredentials, String credentialName, String propertyName)
+    private static Properties updateConnectionProperty(Properties connectionProperties, CredentialProvider credentialProvider, JdbcIdentity identity)
```

Comment:
`updateConnectionProperty` -> `getConnectionProperties`. There is no need for this method to be static.